contracts/governance/Staking/Staking.sol

```diff
@@ -298,21 +298,120 @@ contract Staking is
     }
 
     /**
-     * @notice Withdraw tokens for vesting contract.
-     * @param vesting The address of Vesting contract.
+     * @notice Withdraw the given amount of tokens.
+     * @param amount The number of tokens to withdraw.
+     * @param until The date until which the tokens were staked.
      * @param receiver The receiver of the tokens. If not specified, send to the msg.sender
-     * @dev Can be invoked only by whitelisted contract passed to governanceWithdrawVesting.
+     * @param vestingConfig The vesting config.
+     * @dev VestingConfig struct intended to avoid stack too deep issue, and it contains this properties:
+        address vestingAddress; // vesting contract address
+        uint256 startDate; //start date of vesting
+        uint256 endDate; // end date of vesting
+        uint256 cliff; // cliff. After this time period the tokens begin to unlock.
+        uint256 duration; // The duration. After this period all tokens will have been unlocked.
+        address tokenOwner; // The owner of the vested tokens.
+     * @dev Only used to governance direct withdrawal.
      * */
-    function governanceWithdrawVesting(address vesting, address receiver)
-        public
-        onlyAuthorized
-        whenNotFrozen
-    {
-        vestingWhitelist[vesting] = true;
-        ITeamVesting(vesting).governanceWithdrawTokens(receiver);
-        vestingWhitelist[vesting] = false;
+    function _governanceWithdrawDirect(
```

Comment:
because this direct withdrawal is from vesting contracts only
pls append or insert `Vesting` in all governance direct withdrawal func names for clarity
```suggestion
    function _governanceWithdrawVestingDirect(
```

Reply:
Fixed in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/c4c43c22e2dc30a936960e01e9a8acd6e2053616)